### ISSUE TITLE
Visualize the Gradients in the discriminator on the Tensor-board  

### ISSUE THREAD

**Issue by shamanez** [uidilr/gail_ppo_tf#13]
Hi ,
I think it is useful to visualize gradients in discriminator. Instead of using Adam optimizer in the given way I used,

`   optimizer = tf.train.AdamOptimizer()
    grads = optimizer.compute_gradients(cross_entropy)
    train_step = optimizer.apply_gradients(grads)
    for index, grad in enumerate(grads):
        tf.summary.histogram("{}-grad".format(grads[index][1].name), grads[index])`

**Comment by uidilr** [comment]
Does enumeration of grads (tensorflow variables) works ?
I think tensorflow variables are not iterable.
```python 
for index, grad in enumerate(grads):
    tf.summary.histogram("{}-grad".format(grads[index][1].name), grads[index])
```

**Comment by shamanez** [comment]
yeah it worked.   Sorry for the mistake I used this 

gradients = optimizer.compute_gradients(loss, var_list=pi_trainable)


_

**Comment by uidilr** [comment]
Okay.
Could you make the Pull Request with these codes?

**Comment by shamanez** [comment]
I modified the code to visualize layer weights, activations, and gradients in the censor board. I will pull request as soon as possible just after I tested it bit more.
